pandas/core/arrays/period.py

```diff
@@ -644,12 +649,19 @@ def astype(self, dtype, copy: bool = True):
             return self.asfreq(dtype.freq)
         return super().astype(dtype, copy=copy)
 
-    def searchsorted(self, value, side="left", sorter=None) -> np.ndarray:
-        value = self._validate_searchsorted_value(value).view("M8[ns]")
+    def searchsorted(
+        self,
+        value: ArrayLike | object,
+        side: Literal["left", "right"] = "left",
+        sorter: NumpySorter = None,
+    ) -> np.ndarray:
+        npvalue = cast(
+            np.ndarray, self._validate_searchsorted_value(value).view("M8[ns]")
+        )
```

Comment:
can we avoid the cast with the typing of `_validate_searchsorted_value`

Reply:
removed in next commit.